src/screens/WalletConnectApprovalSheet.tsx

```diff
@@ -211,11 +211,19 @@ export function WalletConnectApprovalSheet() {
    * v2.
    */
   const approvalNetworkInfo = useMemo(() => {
-    const chain = useBackendNetworksStore.getState().getDefaultChains()[approvalChainId || ChainId.mainnet];
+    const chain = useBackendNetworksStore.getState().getDefaultChains()[approvalChainId];
+    if (!chain) {
+      return {
+        chainId: approvalChainId,
+        color: colors.alpha(colors.blueGreyDark, 0.3),
+        name: useBackendNetworksStore.getState().getChainsLabel()[approvalChainId],
+      };
+    }
+
     const nativeAsset = useBackendNetworksStore.getState().getChainsNativeAsset()[chain.id];
     return {
       chainId: chain.id,
-      color: isDarkMode ? nativeAsset.colors.primary : nativeAsset.colors.fallback || nativeAsset.colors.primary,
+      color: isDarkMode ? nativeAsset?.colors.primary : nativeAsset?.colors.fallback || nativeAsset?.colors.primary,
```

Comment:
Are there cases where we have a chain's data but not its native asset?

Reply:
don't believe so, nope

Reply:
Doesn't really matter, but if that's the case then you shouldn't need those optionals on the nativeAsset given the early return you added.